File: frontend/src/api/rules.ts
import { apiClient } from './client'

export interface Rule {
  id: string
  name: string
  category_id?: string
  category_name?: string
  version: number
  status: 'DRAFT' | 'VALIDATED' | 'PUBLISHED'
  effective_from?: string
  created_at: string
  rule_data: Record<string, unknown>
}

export interface RuleListResponse {
  data: Rule[]
  total: number
}

export async function listRules(): Promise<RuleListResponse> {
  const { data } = await apiClient.get<RuleListResponse>('/admin/rules')
  return data
}

export async function getRule(id: string): Promise<Rule> {
  const { data } = await apiClient.get<Rule>(`/admin/rules/${id}`)
  return data
}

export async function createRule(payload: Partial<Rule>): Promise<Rule> {
  const { data } = await apiClient.post<Rule>('/admin/rules', payload)
  return data
}

export async function validateRule(payload: Partial<Rule>): Promise<{ valid: boolean; errors?: string[] }> {
  const { data } = await apiClient.post('/admin/rules/validate', payload)
  return data
}

export async function publishRule(id: string): Promise<Rule> {
  const { data } = await apiClient.post<Rule>(`/admin/rules/${id}/publish`)
  return data
}

File: backend/app/schemas/rules.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, Dict, Any
from datetime import datetime
import uuid

class LMPCRuleBase(BaseModel):
    name: str
    category_id: Optional[uuid.UUID] = None
    rule_data: Dict[str, Any]
    effective_from: Optional[datetime] = None

class LMPCRuleCreate(LMPCRuleBase):
    pass

class LMPCRuleOut(LMPCRuleBase):
    id: uuid.UUID
    version: int
    status: str
    category_name: Optional[str] = None
    created_at: datetime
    
    model_config = ConfigDict(from_attributes=True)

class RuleValidationResult(BaseModel):
    valid: bool
    errors: Optional[list[str]] = None
